### ISSUE TITLE
Simple theme changes (logo, colors, etc)

### ISSUE THREAD

**Issue by sion42x** [openfun/richie#1534]
I am new to Richie, and built it into my Tutor OpenEDX project with https://github.com/overhangio/tutor-richie. I've gone through all the docs, folders, etc. of richie and have figured out the CMS usage, but can't for the life of me figure out how to do simple things like replace the logo, change the CSS, etc. 

The documentation talks about how to build a dev server, templates, etc. but doesn't actually describe how to recompile and reload the site, and perhaps I don't know enough about Django/React to do it. Since I already replaced files in /richie/src/richie/apps/core/static/richie/images etc. I feel like there's something super simple that I'm missing here!

**Comment by sampaccoud** [comment]
@sion42x the fast answer is: we are missing documentation. Richie installs as a third party app in your own Django project. You can take example on what we do for our own sites at https://github.com/openfun/richie-site-factory. There are examples of overriding css, Html or even React templates.

**Comment by sion42x** [comment]
Thanks @sampaccoud, I don't feel **as** bad now 😄 

I've been able to do that on my local dev machine without issues and it's pretty wonderful. That said, I still don't see how to promote what I've changed to a running production richie instance (in my case, running inside tutor-richie). 

Once I can get that figured out I'd be happy to document "Making your first frontend change" or something like that.

**Comment by sampaccoud** [comment]
@sion42x I'm not sure I understand. Your Tutor is an OpenEdX instance in which a `tutor-richie` module is running. Your Richie is a separate project, running on a separate instance. We do not impose a way to host Django, Richie installs in your DjangoCMS app, and you can find a lot of documentation on ways to host Django in production. 
That being said, we at FUN run Django in production using Docker and Kubernetes. The Docker images are built in https://github.com/openfun/richie-site-factory and the deployment to Kubernetes is done with [Arnold](https://github.com/openfun/arnold) via the [richie app](https://github.com/openfun/arnold/tree/master/apps/richie). You can refer to these repositories for inspiration or as boilerplates for your own deployments.

**Comment by sion42x** [comment]
Thanks @sampaccoud, I guess it's just a case of me not understanding Docker and the frontend. I think we're describing two different things. 

As you say, I'm using tutor-richie (`pip install tutor-richie`, then `tutor plugins enable richie`), which runs in a docker container and automatically enables a subdomain of my OpenEDX domain (courses.mydomain.com) that also is the default redirect location for the top level domain. I did NOT run the create demo site function. Django Admin is fully configurable from the admin page. What I can't seem to figure out how to do is modify the default front-end UI that is running in the richie docker container (which has the "your logo here" up top and standard richie style). 

I think the problem is me just not understanding docker/react enough. I'll keep playing with it unless you have any ideas.

**Comment by sampaccoud** [comment]
@sion42x I'd be happy to make a p2p call to better understand your situation as I'm not familar with Tutor and the plugin.
Maybe @regisb you can also give us your 2 cents?

**Comment by regisb** [comment]
Let's ask the question differently: if you were running Richie as a Django app natively, on the host, inside your own Django CMS project, how would you theme Richie? Which files would you modify? There is currently no simple, user-friendly mechanism in place to modify the Richie theme from the tutor-richie plugin, but we can work on a solution.

**Comment by sampaccoud** [comment]
This is what is done in https://github.com/openfun/richie-site-factory.

**Comment by sion42x** [comment]
> Let's ask the question differently: if you were running Richie as a Django app natively, on the host, inside your own Django CMS project, how would you theme Richie? Which files would you modify? There is currently no simple, user-friendly mechanism in place to modify the Richie theme from the tutor-richie plugin, but we can work on a solution.

From looking through https://github.com/openfun/richie-site-factory and the docker image, I've been attempting changes to /richie/src/richie/apps/core/static/richie/images and other locations (which automatically revert). I'd assume at this point the "friendly" way to do it for tutor-richie is to make a function similar to the tutor openedx theming process (make changes in a local location, tutor images build richie, tutor start -d with the appropriate changes to the Dockerfile). 

In the meantime, it seems the way would be to clone the tutor-richie image, modify, and use my own image in production. Does that sound right @regisb?

Depending on the level of effort involved I may be willing to fund a solution for this in tutor-richie. I'll reach out to @regisb via email on that.

**Comment by sion42x** [comment]
I was able to make some progress on this finally, and have indeed chalked it up to me not understanding Docker well enough. Here's what I did for posterity:

1. Built a local dev environment
2. Modified files in frontend/scss and richie/apps
3. Copied those folders to my production server under the directory $(tutor config printroot)/env/plugins/richie/build/richie/richie-theme
4. Edited the Dockerfile at $(tutor config printroot)/env/plugins/richie/build/richie/Dockerfile to add two COPY commands in the base image block:
```
#--------- Base image with cloned repo
FROM python:3.7-bullseye as base

RUN apt update \
    && apt install -y git \
    && rm -rf /var/lib/apt/lists/*

# Clone repo
ARG RICHIE_REPOSITORY=https://github.com/openfun/richie.git
ARG RICHIE_VERSION=v2.8.2
RUN git clone $RICHIE_REPOSITORY --branch $RICHIE_VERSION --depth 1 /richie
COPY ./richie-theme/src/frontend/scss /richie/src/frontend/scss
COPY ./richie-theme/src/richie/apps /richie/src/richie/apps
```

5. Ran `tutor images build richie`
6. Ran `tutor local start -d`

I don't know if this was the best way, but it WORKED!

**Comment by sampaccoud** [comment]
@sion42x Thank you for your contribution! I discussed with @regisb and discovered that the [tutor-richie](https://github.com/overhangio/tutor-richie) plugin also ran the richie instance, and not just a connector in OpenEdX. That's why we did not understand each other at first, sorry :sweat_smile: 

France Université Numérique will finance an improvement in the tutor-richie plugin so that customization is easier and documented. Basically what you did is the way to do it and we will bake this mechanism in the plugin.
